driver-core/src/main/java/com/datastax/driver/core/DataType.java

```diff
@@ -188,7 +188,7 @@ static DataType decode(ByteBuf buffer, ProtocolVersion protocolVersion, CodecReg
                     DataType fieldType = decode(buffer, protocolVersion, codecRegistry);
                     fields.add(new UserType.Field(fieldName, fieldType));
                 }
-                return new UserType(keyspace, type, fields, protocolVersion, codecRegistry);
+                return new UserType(keyspace, type, true, fields, protocolVersion, codecRegistry);
```

Comment:
I think the reason it is done this way is so the behavior doesn't change from the way it worked.  Previously a `UserType` would always be considered frozen, so in the case of parsing the type back from result metadata where there is no way to tell, returning true will provide similar behavior as it did in the past.